webserver/views/datasets.py

```diff
@@ -286,36 +298,65 @@ def create_service():
 def import_csv():
     form = forms.DatasetCSVImportForm()
     if form.validate_on_submit():
+        description, classes = _parse_dataset_csv(request.files[form.file.name])
         dataset_dict = {
             "name": form.name.data,
-            "description": form.description.data,
-            "classes": _parse_dataset_csv(request.files[form.file.name]),
+            "description": description if description else form.description.data,
+            "classes": classes,
             "public": True,
         }
         try:
             dataset_id = db.dataset.create_from_dict(dataset_dict, current_user.id)
         except dataset_validator.ValidationException as e:
             raise BadRequest(str(e))
         flash.info("Dataset has been imported successfully.")
-        return redirect(url_for(".view", id=dataset_id))
+        return redirect(url_for(".view", dataset_id=dataset_id))
```

Comment:
There was a bug here which caused this page to not load... You probably should have encountered this if you tested this import page. Did you see the issue?

Reply:
Yes I did see that issue, I should have brought it up - apologies. One thing that I noted, this issue only happens when importing CSV files on localhost - importing CSV on acousticbrainz.org does not experience the redirect error. Do you know why this is?

Reply:
We've not released the change that introduced this bug yet, which is why it's not present on acousticbrainz.org. If you see a problem like this and you're not sure if it's a problem or not, you can always ask us about it, in IRC, or a ticket comment, or open a new ticket.

Reply:
Okay makes sense, thanks for clearing that up! If I'm confused about a problem like that I'll be sure to ask about it along the way.